src/main/java/net/dv8tion/jda/core/entities/SelfUser.java

```diff
@@ -54,6 +54,43 @@
      */
     String getEmail() throws AccountTypeException;
 
+    // TODO: find out if this is the real meaning of the field or if it has another purpose
+    /**
+     * Shows whether there has ever been a mobile app connected to this account.
+     * <br><b>NOTE:</b> this is a {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT} method only!
+     *
+     * @throws AccountTypeException
+     *         If this method is called when {@link net.dv8tion.jda.core.JDA#getAccountType() JDA#getAccountType()} does not return
+     *         {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT}. E.g: If the logged in account isn't a Client account!
+     *
+     * @return {@code true} if the account is linked with a mobile app, otherwise {@code false}
+     */
+    boolean isMobile() throws AccountTypeException;
+
+    /**
+     * The Discord Nitro status of this account.
+     * <br><b>NOTE:</b> this is a {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT} method only!
+     *
+     * @throws AccountTypeException
+     *         If this method is called when {@link net.dv8tion.jda.core.JDA#getAccountType() JDA#getAccountType()} does not return
+     *         {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT}. E.g: If the logged in account isn't a Client account!
+     *
+     * @return The Discord Nitro status of the currently logged in account.
+     */
+    boolean isPremium() throws AccountTypeException;
+
+    /**
+     * Used to get the phone number of the currently logged in account if a phone number has been attached to it.
+     * <br><b>NOTE:</b> this is a {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT} method only!
+     *
+     * @throws AccountTypeException
+     *         If this method is called when {@link net.dv8tion.jda.core.JDA#getAccountType() JDA#getAccountType()} does not return
+     *         {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT}. E.g: If the logged in account isn't a Client account!
+     *
+     * @return The phone of the currently logged in account or null if there's no number associated
+     */
+    String getPhoneNumber() throws AccountTypeException;
```

Comment:
Why do you specifically throw an `AccountTypeException` for these methods? I would suggest to actually just return w/e we parsed and document that for bots the value will most likely be false or null. It might be possible that discord will make some of these available to bots in the future (unlikely) so we should just go with that, imo.

Reply:
I chose this because we already have the same be behavior for `getEmail()`. I also highly doubt that any of these will be available for bots anytime in the future.